internal/util/helpers_test.go

```diff
@@ -4,18 +4,765 @@
+	err := os.MkdirAll(filepath.Dir(filePath), 0750)
+
+	if err != nil {
+		t.Fatalf("error creating directory: %v", err)
+	}
+
+	err = os.WriteFile(filePath, []byte(`{"access_token":"test_access_token","refresh_token":"test_refresh_token","access_token_expires_at":1234567890}`), 0600)
+	if err != nil {
+		t.Fatalf("error writing credentials to file: %v", err)
+	}
+
+	err = util.RemoveCredentials()
+	if err != nil {
+		t.Fatalf("expected no error, got %v", err)
+	}
+
+	// Verify the file is removed
+	if _, err := os.Stat(filePath); !os.IsNotExist(err) {
+		t.Errorf("expected file to be removed, but it still exists")
+	}
+}
+
+// TestRefreshCredentials tests the RefreshCredentials function
+func TestRefreshCredentials(t *testing.T) {
+	t.Parallel()
+	// Create a temporary directory
+	testDir := t.TempDir()
+
+	SetEnvVar(t, "XDG_CONFIG_HOME", testDir)
+	tests := []struct {
+		name           string
+		refreshToken   string
+		issuerUrl      string
+		clientId       string
+		responseBody   string
+		expectedError  string
+		expectedResult util.OpenIdCredentials
+	}{
+		{
+			name:         "Successful refresh with local server",
+			refreshToken: "valid_refresh_token",
+			issuerUrl:    "http://localhost:8081",
+			clientId:     "minder-cli",
+			responseBody: `{"access_token":"new_access_token","refresh_token":"new_refresh_token","expires_in":3600}`,
+			expectedResult: util.OpenIdCredentials{
+				AccessToken:          "new_access_token",
+				RefreshToken:         "new_refresh_token",
+				AccessTokenExpiresAt: time.Now().Add(3600 * time.Second),
+			},
+		},
+		{
+			name:          "Error fetching new credentials (responseBody is missing) rwith local server",
+			refreshToken:  "valid_refresh_token",
+			issuerUrl:     "http://localhost:8081",
+			clientId:      "minder-cli",
+			expectedError: "error unmarshaling credentials: EOF",
+		},
+		{
+			name:          "Error unmarshaling credentials with local server",
+			refreshToken:  "valid_refresh_token",
+			issuerUrl:     "http://localhost:8081",
+			clientId:      "minder-cli",
+			responseBody:  `invalid_json`,
+			expectedError: "error unmarshaling credentials: invalid character 'i' looking for beginning of value",
+		},
+		{
+			name:          "Error refreshing credentials with local server",
+			refreshToken:  "valid_refresh_token",
+			issuerUrl:     "http://localhost:8081",
+			clientId:      "minder-cli",
+			responseBody:  `{"error":"invalid_grant","error_description":"Invalid refresh token"}`,
+			expectedError: "error refreshing credentials: invalid_grant: Invalid refresh token",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+				w.Header().Set("Content-Type", "application/json")
+				fmt.Fprintln(w, tt.responseBody)
+			}))
+			defer server.Close()
+
+			parsedURL, _ := url.Parse(server.URL)
+			tt.issuerUrl = parsedURL.String()
```

Comment:
Why do this, rather than simply:

```
tt.issuerUrl = server.URL
```

(or simply use `server.URL` below, and don't make the `issuerUrl` a field in the testcase struct, since it is never used...)

Reply:
yes. I can simplify as suggested.